docs/zkapps/how-to-write-a-zkapp-ui.mdx

```diff
@@ -226,3 +225,27 @@ href="https://docs.aurowallet.com/general/reference/api-reference/mina-provider-
 ">reference docs</a>.
 
 :::
+
+### Displaying assertion exceptions in your UI
+
+If an assertion error occurs while a user is interacting with your zkApp, you may want capture the assertion error and display a helpful message for the user in your UI.
+
+1. Capture errors that occur when a user interacts with a method on your smart contract in a try catch statement.
+2. Filter by the error message that was created.
+3. Display a helpful error message for the user in your UI based on what assertion exception was captured. This could be a modal or a message in your UI.
+
+```ts
+try {
+  YourSmartContract.yourMethod();
+} catch (err) {
+  let uiErrorMessage;
+  switch (err.message) {
+    case 'Overflow': // The assertion error message thrown by invoking YourSmartContract.yourMethod when there is an insufficent balance.
```

Comment:
Wasn't your previous case text clearer?

Reply:
It was clearer but `INSUFFICIENT_BALANCE` is not an actual assertion error that is thrown.

Reply:
I assumed it was a custom error message, which is why I suggested that we include a line showing people an assertion with this custom error message. 

IMO, Overflow is not clear and this highlights it needs to be changed elsewhere; seems better to not use it here.

Reply:
I struggled to find a common assertion error a zkApp user might encounter that is relatively clear to show as an example. This highlights the deeper problem we have with error messages. This ambiguity can be mitigated to a point when we have a document mapping the errors to their causes that developers can reference.

Reply:
Then we really need to prioritize that! I'll add an issue to SnarkyJS repo

Meanwhile, can we use a custom error message here please? So it's clear to people.